File: CSharpCodeBase/components/controlsystem/enemyactions/actionpush.lua
local Action = require 'components.controlsystem.action'
local Class = require 'utils.hump.class'

local ActionPush = Class({__includes=Action, init = function (self, entity, targetPosition)
      Action.init(self, 4, false, 'push')
      self.entity = entity
      self.targetPosition = targetPosition
end})

function ActionPush:OnStartRunning()
  --Action.OnStart(self)
  self.entity.mover:Stop()
  self.entity.mecanim:SetFloat("action_type", 1)
  self.entity.mecanim:ResetTrigger("action")
  self.entity.mecanim:ForceSetState("Hit")
  self.dv = self.entity:GetPosition() - self.targetPosition
  self.dv.y = 0
  self.dv:Normalize()
  --print(self.targetPosition.x .. ',' .. self.targetPosition.z)
  self.isHitStarted = false
end

function ActionPush:Update()
  if not self.isHitStarted then
    self.isHitStarted = self.entity.mecanim:CheckStateName("Hit")
  end
  
  return self.isHitStarted and not self.entity.mecanim:CheckStateName("Hit")
end

function ActionPush:FixedUpdate()
  self.entity.mover:LookAt(self.targetPosition)
  local speedCurve = self.entity.mecanim:GetFloat('speed_curve')
  --self.entity.mover:SetInput(self.dv * speedCurve)
  RigidbodyUtils.MoveNotRotate(self.entity.rigidbody, self.dv * speedCurve * 10)
end

function ActionPush:OnStopRunning()
  self.entity.mover:Stop()
end

return ActionPush

File: CSharpCodeBase/components/controlsystem/enemyactions/actionpush.cs
using System;
 
 using System.Collections.Generic;
 
 using System.Linq;
 
 using System.Text;
 
 using UnityEngine;
 
 using System.Collections;
 
 using MainGame.core;
 
 using UnityEngine.EventSystems;
 
 namespace MainGame{
 public class actionpush :action{
 public void init (self, entity, targetPosition);
       Action.init(self, 4, false, "push");
       this.entity = entity;
       this.targetPosition = targetPosition;
 }})
 public void OnStartRunning(){
   //Action.OnStart(self)
   this.entity.mover:Stop();
   this.entity.mecanim:SetFloat("action_type", 1);
   this.entity.mecanim:ResetTrigger("action");
   this.entity.mecanim:ForceSetState("Hit");
   this.dv = this.entity:GetPosition() - this.targetPosition;
   this.dv.y = 0;
   this.dv:Normalize();
   //print(this.targetPosition.x .. "," .. this.targetPosition.z)
   this.isHitStarted = false;
 }
 public void Update(){
   if(not this.isHitStarted  ){ 
     this.isHitStarted = this.entity.mecanim:CheckStateName("Hit");
   }
   
   return this.isHitStarted  &&  not this.entity.mecanim:CheckStateName("Hit");
 }
 public void FixedUpdate(){
   this.entity.mover:LookAt(this.targetPosition);
   var speedCurve = this.entity.mecanim:GetFloat("speed_curve");
   //this.entity.mover:SetInput(this.dv * speedCurve)
   RigidbodyUtils.MoveNotRotate(this.entity.rigidbody, this.dv * speedCurve * 10);
 }
 public void OnStopRunning(){
   this.entity.mover:Stop();
 }
 }}